test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -222,3 +224,67 @@ timestamp_manager::validate_read_timestamp(session_simulator *session, const uin
 
     return (0);
 }
+
+/* Validate the commit timestamp. */
+int
+timestamp_manager::validate_commit_timestamp(
+  session_simulator *session, const uint64_t commit_ts) const
+{
+    uint64_t prepare_ts = session->get_prepare_timestamp();
+    /*
+     * We cannot set the commit timestamp to be earlier than the fist commit timestamp when setting
+     * the commit timestamp multiple times within a transaction.
+     */
+    uint64_t first_commit_ts = session->get_first_commit_timestamp();
+    if (first_commit_ts != 0 && commit_ts < first_commit_ts) {
+        WT_SIM_RET_MSG(EINVAL,
+          "commit timestamp " + std::to_string(commit_ts) +
+            " older than the first commit timestamp " + std::to_string(first_commit_ts) +
+            " for this transaction");
+    }
+
+    /*
+     * For a non-prepared transactions the commit timestamp should not be less or equal to the
+     * oldest and/or stable timestamp.
+     */
+    connection_simulator *conn = &connection_simulator::get_connection();
+    uint64_t oldest_ts = conn->get_oldest_ts();
+    if (oldest_ts != 0 && commit_ts < oldest_ts) {
+        WT_SIM_RET_MSG(EINVAL,
+          "commit timestamp " + std::to_string(commit_ts) + "is less than the oldest timestamp " +
+            std::to_string(oldest_ts));
+    }
+
+    uint64_t stable_ts = conn->get_stable_ts();
+    if (stable_ts != 0 && commit_ts <= stable_ts) {
+        WT_SIM_RET_MSG(EINVAL,
+          "commit timestamp " + std::to_string(commit_ts) + "must be after the stable timestamp " +
+            std::to_string(stable_ts));
+    }
+
+    /* The commit timestamp must be greater than the latest active read timestamp. */
+    uint64_t latest_active_read = conn->get_latest_active_read();
+    if (latest_active_read >= commit_ts)
+        WT_SIM_RET_MSG(EINVAL,
+          "commit timestamp " + std::to_string(commit_ts) +
+            "must be after all active read timestamps " + std::to_string(latest_active_read));
+
+    /*
+     * For a prepared transaction, the commit timestamp should not be less than the prepare
+     * timestamp. Also, the commit timestamp cannot be set before the transaction has actually been
+     * prepared.
+     *
+     * If the commit timestamp is less than the oldest timestamp and the transaction is configured
+     * to roundup timestamps of a prepared transaction, then we will roundup the commit timestamp to
+     * the prepare timestamp of the transaction.
+     */
+    if (session->has_prepare_timestamp()) {
```

Comment:
I'll leave this as a separate if statement for now since further logic needs to be added as part of the prepare_ts ticket.